coin_prices_from_coingecko/output.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/blocky/as-demo/as"
+)
+
+type Result struct {
+	Success bool
+	Error   string
+	Value   any
+}
+
+func (r Result) jsonMarshalWithError(err error) []byte {
```

Comment:
If we are going with a design in which all functions are expolrted, consider exporting this function as well.

Reply:
updating